launcher/sdw_updater_gui/UpdaterApp.py

```diff
@@ -158,6 +189,26 @@ def exit_launcher(self):
         sys.exit()
 
 
+def _is_netcheck_successful() -> bool:
+    """
+    Helper function to assess network connectivity before launching updater.
+
+    Assess network connectivity by checking connection status (via nmcli) in
+    sys-net.
+    """
+    command = b"nmcli networking connectivity check"
+
+    if not Util.get_qubes_version():
+        logger.error("QubesOS not detected, cannot check network.")
+        return False
+    try:
+        result = subprocess.check_output(["qvm-run", "sys-net", command])
+        return result is not None and result.decode("utf-8") == "full"
+    except subprocess.CalledProcessError:
+        logger.error("{} (connectivity check) failed".format(command.decode("utf-8")))
```

Comment:
Logging is good! More logging! However, I wasn't able to find this message in the logs during my testing. Logging the actual return value would be helpful during debugging sessions. From [the docs](https://docs.python.org/3/library/subprocess.html#subprocess.check_output): 

> If the return code was non-zero it raises a CalledProcessError. The CalledProcessError object will have the return code in the returncode attribute and any output in the output attribute.

So I suggest something like:

```
except subprocess.CalledProcessError as e:
    logger.error("{} (connectivity check) failed; state reported as '{}'".format(command.decode("utf-8"), e.output))
```

Make sure you can find that string in the on-disk log files when a failure event is hit.

Reply:
Thank you -addressed in d5f2d1d (Note this log message will not be encountered in a no-network scenario; in that case nmcli will still return 0 but the error log will read `Network connectivity check failed`).